pkg/applyconfigurations/applyconfiguration_integration_test.go

```diff
@@ -0,0 +1,144 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applyconfigurations
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+type outputToMap map[string]*outputFile
+
+// Open implements genall.OutputRule.
+func (m outputToMap) Open(_ *loader.Package, path string) (io.WriteCloser, error) {
+	if _, ok := m[path]; !ok {
+		m[path] = &outputFile{}
+	}
+	return m[path], nil
+}
+
+type outputFile struct {
+	contents []byte
+}
+
+func (o *outputFile) Write(p []byte) (int, error) {
+	o.contents = append(o.contents, p...)
+	return len(p), nil
+}
+
+func (o *outputFile) Close() error {
+	return nil
+}
+
+var _ = Describe("CRD Generation From Parsing to CustomResourceDefinition", func() {
+	It("should be able to verify generated ApplyConfiguration types for the CronJob schema", func() {
+		By("switching into testdata to appease go modules")
+		cwd, err := os.Getwd()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(os.Chdir("./testdata/cronjob")).To(Succeed()) // go modules are directory-sensitive
+		defer func() { Expect(os.Chdir(cwd)).To(Succeed()) }()
+
+		output := make(outputToMap)
+
+		By("initializing the runtime")
+		optionsRegistry := &markers.Registry{}
+		Expect(optionsRegistry.Register(markers.Must(markers.MakeDefinition("crd", markers.DescribesPackage, crd.Generator{})))).To(Succeed())
+		// Add the applyconfigurations generator but set it to verify only.
+		// This allows us to check if there's a diff between the checked in generated data and what it would generate now.
+		Expect(optionsRegistry.Register(markers.Must(markers.MakeDefinition("apply", markers.DescribesPackage, Generator{})))).To(Succeed())
+		rt, err := genall.FromOptions(optionsRegistry, []string{
+			"crd", // Run another generator first to make sure they don't interfere; see also: the comment on cronjob_types.go:UntypedBlob
+			"apply",
+		})
+		Expect(err).NotTo(HaveOccurred())
+		rt.OutputRules = genall.OutputRules{Default: output}
+
+		By("running the generator and checking for errors")
+		hadErrs := rt.Run()
+
+		By("checking for errors")
+		Expect(hadErrs).To(BeFalse(), "Generator should run without errors")
+
+		filesInMaster := make(map[string][]byte)
+		masterFileNames := sets.New[string]()
+		cronJobFS := os.DirFS(".")
+		masterPath := "applyconfiguration-master"
```

Comment:
Because I can't change the directory to which we are generating the assets currently, this is what I came up with, create a copy that we keep checked in and then diff against that, if there are genuine changes we can update both copies.

Will look at gengo to see if there's a backwards compatible way to allow it to output to different places like `OutputRule` does for these generators